tensorflow/core/ops/math_ops.cc

```diff
@@ -131,90 +131,102 @@ tf.complex_abs(x) ==> [5.25594902, 6.60492229]
 
 // Declares cwise unary operations signature: 't -> 't
 #define UNARY()                      \
-  Input("x: T").Output("y: T").Attr( \
-      "T: {half, float, double, int32, complex64, int64}")
+  Input("x: T").Output("y: T")
```

Comment:
We definitely don't want trigonometric functions of integers.  I think the macros should be

* `UNARY`: Explicit list of types (what you did)
* `UNARY_ALL`: the current list of types
* `UNARY_REAL`: half, float, double (not sure why you stuck int64 in their)
* `UNARY_COMPLEX`: half, float, double, complex64, complex128

where the last three macros are implemented in terms of the first.